pep-0680.rst

```diff
@@ -0,0 +1,501 @@
+understanding the amount of breakage we could expect if we used the ``toml``
+name for the standard library module, as well as to better understand the design
+space. Note that this list might not be exhaustive.
+
+#. This PEP currently proposes not to include a write API. That is, there will
+   be no equivalent of ``toml.dump`` or ``toml.dumps``.
+
+   Discussed at `<Including an API for writing TOML_>`_.
+
+   If we included a write API, it would be relatively simple to convert most
+   code that uses ``toml`` to use the API proposed in this PEP (acknowledging
+   that that is very different from a compatible API).
+
+   A significant fraction of ``toml`` users rely on this.
+
+#. Different first argument of ``toml.load``
+
+   ``toml.load`` has the following signature:
+
+   .. code-block::
+
+       def load(
+           f: Union[SupportsRead[str], str, bytes, list[PathLike | str | bytes]],
+           _dict: Type[MutableMapping[str, Any]] = ...,
+           decoder: TomlDecoder = ...,
+       ) -> MutableMapping[str, Any]: ...
+
+   This is pretty different from the first argument proposed in this PEP: ``SupportsRead[bytes]``.
+
+   Recapping the reasons for this, previously mentioned at
+   `<Types accepted by the first argument of tomllib.load_>`_:
+
+   * Allowing passing of paths (and lists of path-like objects, ignoring missing
+     files and merging the documents into a single object) is inconsistent with
+     other similar functions in the standard library.
+   * Using ``SupportsRead[bytes]`` allows us to a) ensure utf-8 is the encoding used,
+     b) avoid incorrectly parsing single carriage returns as valid TOML due to
+     universal newlines. TOML specifies file encoding and valid newline
+     sequences, and hence is simply stricter format than what text file objects
+     represent.
+
+   A significant fraction of ``toml`` users rely on this.
+
+#. Errors
+
+   ``toml`` raises ``TomlDecodeError`` vs the proposed PEP 8 compliant
+   ``TOMLDecodeError``.
+
+   A significant fraction of ``toml`` users rely on this.
+
+#. ``toml.load[s]`` accepts a ``_dict`` argument
+
+   Discussed at `<Controlling the type of mappings returned by tomllib.load[s]_>`_.
+
+   As discussed, almost all usage consists of ``_dict=OrderedDict``, which is
+   not necessary in Python 3.7 and later.
+
+#. ``toml.load[s]`` support an undocumented ``decoder`` argument
+
+   It seems the intended use case is for an implementation of comment
+   preservation. The information recorded is not sufficient to roundtrip the
+   TOML document preserving style, the implementation has known bugs, the
+   feature is undocumented and I could only find one instance of its use on
+   https://grep.app.
+
+   The ``toml.TomlDecoder`` interface exposed is not simple, containing nine methods.
+   See `here <https://github.com/uiri/toml/blob/3f637dba5f68db63d4b30967fedda51c82459471/toml/decoder.pyi#L36>`__.
```

Comment:
Using `here` or `click` for text links is poor metadata and accessibility

Reply:
Fixed [here](https://github.com/python/peps/pull/2223)

(haha :smile: )